### PR TITLE
tests: Add EvalScript(...) fuzzing harness

### PR DESCRIPTION
**Pull request by practicalswift**
Add `EvalScript(...)` fuzzing harness.

To test this PR:

We can run `contrib/devtools/test_fuzzing_harnesses.sh` (#17000) during five seconds to quickly verify that the newly added  fuzz harness seem to hit relevant code regions, that the fuzzing throughput seems reasonable, etc.

`test_fuzzing_harnesses.sh eval 5` runs all fuzzers matching the regexp `eval` giving them five seconds of runtime each.

```
$ CC=clang CXX=clang++ ./configure --enable-fuzz --with-sanitizers=address,fuzzer,undefined
$ make
$ contrib/devtools/test_fuzzing_harnesses.sh eval 5
Testing fuzzer eval_script during 5 second(s)
A subset of reached functions:
        NEW_FUNC[1/24]: 0x557b808742e0 in prevector<28u, unsigned char, unsigned int, int>::indirect_ptr(int) src/./prevector.h:161
        NEW_FUNC[2/24]: 0x557b80875460 in prevector<28u, unsigned char, unsigned int, int>::indirect_ptr(int) const src/./prevector.h:162
        NEW_FUNC[6/9]: 0x557b81acdaa0 in popstack(std::vector<std::vector<unsigned char, std::allocator<unsigned char> >, std::allocator<std::vector<unsigned char, std::allocator<unsigned char> > > >&) src/script/interpreter.cpp:57
        NEW_FUNC[5/16]: 0x557b809f1bf0 in CScriptNum::serialize(long const&) src/./script/script.h:326
        NEW_FUNC[4/6]: 0x557b817c93d0 in CScriptNum::CScriptNum(std::vector<unsigned char, std::allocator<unsigned char> > const&, bool, unsigned long) src/./script/script.h:225
        NEW_FUNC[5/6]: 0x557b817cbb80 in CScriptNum::set_vch(std::vector<unsigned char, std::allocator<unsigned char> > const&) src/./script/script.h:360
        NEW_FUNC[0/11]: 0x557b80a88170 in CHash256::Write(unsigned char const*, unsigned long) src/./hash.h:34
        NEW_FUNC[1/11]: 0x557b80a88270 in CHash256::Finalize(unsigned char*) src/./hash.h:28
        NEW_FUNC[5/11]: 0x557b81affdb0 in CSHA256::CSHA256() src/crypto/sha256.cpp:644
        NEW_FUNC[6/11]: 0x557b81affe80 in (anonymous namespace)::sha256::Initialize(unsigned int*) src/crypto/sha256.cpp:66
        NEW_FUNC[7/11]: 0x557b81b00460 in CSHA256::Write(unsigned char const*, unsigned long) src/crypto/sha256.cpp:649
        NEW_FUNC[8/11]: 0x557b81b009a0 in CSHA256::Finalize(unsigned char*) src/crypto/sha256.cpp:675
        NEW_FUNC[9/11]: 0x557b81b015e0 in CSHA256::Reset() src/crypto/sha256.cpp:692
        NEW_FUNC[10/11]: 0x557b81b01d90 in (anonymous namespace)::sha256::Transform(unsigned int*, unsigned char const*, unsigned long) src/crypto/sha256.cpp:79
        NEW_FUNC[0/1]: 0x557b808cc180 in BaseSignatureChecker::CheckLockTime(CScriptNum const&) const src/./script/interpreter.h:153
        NEW_FUNC[0/2]: 0x557b81ab5640 in CastToBool(std::vector<unsigned char, std::allocator<unsigned char> > const&) src/script/interpreter.cpp:36
        NEW_FUNC[0/1]: 0x557b817c9c30 in CScriptNum::getint() const src/./script/script.h:312
        NEW_FUNC[0/1]: 0x557b81ae1df0 in CScriptNum::operator-=(long const&) src/./script/script.h:298
        NEW_FUNC[0/5]: 0x557b81af5670 in CRIPEMD160::CRIPEMD160() src/crypto/ripemd160.cpp:243
        NEW_FUNC[1/5]: 0x557b81af5740 in (anonymous namespace)::ripemd160::Initialize(unsigned int*) src/crypto/ripemd160.cpp:25
        NEW_FUNC[2/5]: 0x557b81af5b00 in CRIPEMD160::Write(unsigned char const*, unsigned long) src/crypto/ripemd160.cpp:248
        NEW_FUNC[3/5]: 0x557b81af5fa0 in (anonymous namespace)::ripemd160::Transform(unsigned int*, unsigned char const*) src/crypto/ripemd160.cpp:55
        NEW_FUNC[4/5]: 0x557b81af8d60 in CRIPEMD160::Finalize(unsigned char*) src/crypto/ripemd160.cpp:274
        NEW_FUNC[0/16]: 0x557b80857a30 in CScript::operator<<(std::vector<unsigned char, std::allocator<unsigned char> > const&) src/./script/script.h:462
        NEW_FUNC[1/16]: 0x557b80872670 in prevector<28u, unsigned char, unsigned int, int>::insert(prevector<28u, unsigned char, unsigned int, int>::iterator, unsigned char const&) src/./prevector.h:342
        NEW_FUNC[2/16]: 0x557b80872e00 in void prevector<28u, unsigned char, unsigned int, int>::insert<__gnu_cxx::__normal_iterator<unsigned char const*, std::vector<unsigned char, std::allocator<unsigned char> > > >(prevector<28u, unsigned char, unsigned int, int>::iterator, __gnu_cxx::__normal_iterator<unsigned char const*, std::vector<unsigned char, std::allocator<unsigned char> > >, __gnu_cxx::__normal_iterator<unsigned char const*, std::vector<unsigned char, std::allocator<unsigned char> > >) src/./prevector.h:368
        NEW_FUNC[3/16]: 0x557b80873630 in prevector<28u, unsigned char, unsigned int, int>::capacity() const src/./prevector.h:295
        NEW_FUNC[4/16]: 0x557b80874ed0 in void prevector<28u, unsigned char, unsigned int, int>::fill<prevector<28u, unsigned char, unsigned int, int>::const_iterator>(unsigned char*, prevector<28u, unsigned char, unsigned int, int>::const_iterator, prevector<28u, unsigned char, unsigned int, int>::const_iterator) src/./prevector.h:204
        NEW_FUNC[5/16]: 0x557b808cc0f0 in BaseSignatureChecker::CheckSig(std::vector<unsigned char, std::allocator<unsigned char> > const&, std::vector<unsigned char, std::allocator<unsigned char> > const&, CScript const&, SigVersion) const src/./script/interpreter.h:148
        NEW_FUNC[6/16]: 0x557b809edb10 in CScript::operator=(CScript&&) src/./script/script.h:390
        NEW_FUNC[7/16]: 0x557b809f8ec0 in void prevector<28u, unsigned char, unsigned int, int>::insert<prevector<28u, unsigned char, unsigned int, int>::const_iterator>(prevector<28u, unsigned char, unsigned int, int>::iterator, prevector<28u, unsigned char, unsigned int, int>::const_iterator, prevector<28u, unsigned char, unsigned int, int>::const_iterator) src/./prevector.h:368
        NEW_FUNC[8/16]: 0x557b809f9260 in prevector<28u, unsigned char, unsigned int, int>::swap(prevector<28u, unsigned char, unsigned int, int>&) src/./prevector.h:451
        NEW_FUNC[9/16]: 0x557b81ab58c0 in CheckSignatureEncoding(std::vector<unsigned char, std::allocator<unsigned char> > const&, unsigned int, ScriptError_t*) src/script/interpreter.cpp:200
        NEW_FUNC[10/16]: 0x557b81ab6f30 in FindAndDelete(CScript&, CScript const&) src/script/interpreter.cpp:254
        NEW_FUNC[11/16]: 0x557b81acdc20 in CheckPubKeyEncoding(std::vector<unsigned char, std::allocator<unsigned char> > const&, unsigned int, SigVersion const&, ScriptError_t*) src/script/interpreter.cpp:217
        NEW_FUNC[12/16]: 0x557b81ad3890 in IsCompressedOrUncompressedPubKey(std::vector<unsigned char, std::allocator<unsigned char> > const&) src/script/interpreter.cpp:63
        NEW_FUNC[13/16]: 0x557b81ad8830 in CScript::GetOp(prevector<28u, unsigned char, unsigned int, int>::const_iterator&, opcodetype&) const src/./script/script.h:505
        NEW_FUNC[14/16]: 0x557b81ae21a0 in prevector<28u, unsigned char, unsigned int, int>::prevector<prevector<28u, unsigned char, unsigned int, int>::const_iterator>(prevector<28u, unsigned char, unsigned int, int>::const_iterator, prevector<28u, unsigned char, unsigned int, int>::const_iterator) src/./prevector.h:246
        NEW_FUNC[0/1]: 0x557b81ae1a40 in CScriptNum::operator+=(long const&) src/./script/script.h:290
        NEW_FUNC[0/5]: 0x557b81af9760 in CSHA1::CSHA1() src/crypto/sha1.cpp:150
        NEW_FUNC[1/5]: 0x557b81af9830 in (anonymous namespace)::sha1::Initialize(unsigned int*) src/crypto/sha1.cpp:32
        NEW_FUNC[2/5]: 0x557b81af9bf0 in CSHA1::Write(unsigned char const*, unsigned long) src/crypto/sha1.cpp:155
        NEW_FUNC[3/5]: 0x557b81afa090 in (anonymous namespace)::sha1::Transform(unsigned int*, unsigned char const*) src/crypto/sha1.cpp:47
        NEW_FUNC[4/5]: 0x557b81afc5e0 in CSHA1::Finalize(unsigned char*) src/crypto/sha1.cpp:181
        NEW_FUNC[0/1]: 0x557b81ada4f0 in CScriptNum::operator-() const src/./script/script.h:278
        NEW_FUNC[0/1]: 0x557b808cc210 in BaseSignatureChecker::CheckSequence(CScriptNum const&) const src/./script/interpreter.h:158
        NEW_FUNC[0/1]: 0x557b81ab5c00 in IsValidSignatureEncoding(std::vector<unsigned char, std::allocator<unsigned char> > const&) src/script/interpreter.cpp:107
stat::number_of_executed_units: 9728
stat::average_exec_per_sec:     1621
stat::new_units_added:          844
stat::slowest_unit_time_sec:    0
stat::peak_rss_mb:              326
Number of unique code paths taken during fuzzing round: 583

Tested fuzz harnesses seem to work as expected.
```

### CODE REVIEW

**Comment by laanwj** [comment]
Concept ACK

File: src/test/fuzz/fuzz.h [hunk 1]
```diff
@@ -9,6 +9,6 @@
 #include <vector>
 
 
-void test_one_input(std::vector<uint8_t> buffer);
+void test_one_input(const std::vector<uint8_t>& buffer);
```
Comment (MarcoFalke) [inline src/test/fuzz/fuzz.h]: Looks fine to do this change, even if we were to switch to a different class (like https://cs.chromium.org/chromium/src/third_party/libFuzzer/src/utils/FuzzedDataProvider.h?rcl=b9f51dc8c98065df0c8da13c051046f5bab833db), the input data remains read-only.

File: test/fuzz/test_runner.py [hunk 2]
```diff
@@ -100,10 +100,12 @@ def main():
 
 def run_once(*, corpus, test_list, build_dir, export_coverage):
     for t in test_list:
+        corpus_path = os.path.join(corpus, t)
+        os.makedirs(corpus_path, exist_ok=True)
```
Comment (MarcoFalke) [inline test/fuzz/test_runner.py]: See the documentation: "The seed corpus to run on (must contain subfolders for each fuzz target)"

the idea was to give the user feedback when the seeds were missing by accident

File: src/test/fuzz/eval_script.cpp [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+#include <script/interpreter.h>
+#include <test/fuzz/fuzz.h>
+
+#include <limits>
+
+void test_one_input(const std::vector<uint8_t>& buffer)
+{
+    const CScript script(buffer.begin(), buffer.end());
+    for (const auto sig_version : {SigVersion::BASE, SigVersion::WITNESS_V0}) {
+        for (const auto flags : {0U, std::numeric_limits<unsigned int>::max()}) {
```
Comment (MarcoFalke) [inline src/test/fuzz/eval_script.cpp]: Can you explain why the flags are hardcoded (as opposed to be read from the input)?

**Review by maflcko — COMMENTED** [review]
Concept ACK

File: src/test/fuzz/eval_script.cpp (hunk 3 above)
Comment (MarcoFalke) [reply src/test/fuzz/eval_script.cpp]: not sure if this helps, but there is a `static bool IsValidFlagCombination` helper

File: test/fuzz/test_runner.py (hunk 2 above)
Comment (practicalswift) [reply test/fuzz/test_runner.py]: The consequence of that is that Travis fails when adding a new fuzzer (like in this case prior to adding that change). That's not how we want things to work, right? :)

File: src/test/fuzz/fuzz.h (hunk 1 above)
Comment (practicalswift) [reply src/test/fuzz/fuzz.h]: Yes, fuzzing inputs are passed read-only by convention for all in-memory fuzzers I've ever used. And `FuzzedDataProvider` is great :)

**Comment by jonatack** [comment]
Concept ACK. Will try to test this soon.

File: test/fuzz/test_runner.py (hunk 2 above)
Comment (MarcoFalke) [reply test/fuzz/test_runner.py]: I think we do want this (as a reminder to generate seeds and upload them for travis)

Comment (practicalswift) [reply test/fuzz/test_runner.py]: I think that creates a inter-repo dependency that is not warranted: the build for https://github.com/bitcoin/bitcoin should not break due to https://github.com/bitcoin-core/qa-assets not being updated. I suggest moving this checking to a cron job or to https://github.com/bitcoin-core/qa-assets.

Our fuzzing framework is currently in a really really poor state: to fix that we want to make it super easy to add new fuzzers. Removing this inter-repo dependency is a quick way to make it easier to add new fuzzers without any measurable cost: modern coverage-guided fuzzers such as `libFuzzer` usually provide good value even in the absence of a starting corpus.

File: src/test/fuzz/eval_script.cpp (hunk 3 above)
Comment (practicalswift) [reply src/test/fuzz/eval_script.cpp]: It is a trade-off: while reading the flags could increase coverage it would also greatly increase the search space in a way that is not trivial for a fuzzer to reduce intelligently (AFAICT). Based on my reading of how `flags` are affecting the code paths taken in `EvalScript` I came to the conclusion that `{0U, std::numeric_limits<unsigned int>::max()}` seemed like a good way to limit search space while retaining good coverage. Let me know if you can find counter examples of code paths that won't be taken due to this limitation that would warrant increasing the search space :)

Comment (MarcoFalke) [reply src/test/fuzz/eval_script.cpp]: It might give you the same line coverage, but not the same path coverage. E.g. CLTV and CSV are always both enabled or both disabled, but never only one of them is enabled. Do you have any reading suggestions for me that explain your intuition that the fuzzer will conceptually have a harder time covering the search space when the input data is used to decide which code path to take?

Comment (practicalswift) [reply src/test/fuzz/eval_script.cpp]: I'll try to find a link :)  I've now switched to reading `flags` from input using the excellent `FuzzedDataProvider` helper. I'm sure that will work out well too :)

File: test/fuzz/test_runner.py (hunk 2 above)
Comment (practicalswift) [reply test/fuzz/test_runner.py]: I'm now using suppressions in `fuzz/test_runner.py` to resolve this.

File: src/test/fuzz/fuzz.h (hunk 1 above)
Comment (MarcoFalke) [reply src/test/fuzz/fuzz.h]: Could split up this commit, so that other pull request don't conflict with this one?

Comment (MarcoFalke) [reply src/test/fuzz/fuzz.h]: @practicalswift ^ ;)

Comment (practicalswift) [reply src/test/fuzz/fuzz.h]: I've now moved this change to a separate PR to make this PR non-conflicting with others. I hope that was the correct reading of the suggestion :)

File: src/test/fuzz/eval_script.cpp [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+#include <test/fuzz/fuzz.h>
+
+#include <limits>
+
+void test_one_input(const std::vector<uint8_t>& buffer)
+{
+    if (buffer.size() <= sizeof(unsigned int)) {
+        return;
+    }
+    FuzzedDataProvider fuzzed_data_provider(buffer.data(), buffer.size());
```
Comment (MarcoFalke) [inline src/test/fuzz/eval_script.cpp]: style-nit: Looks like this creates another copy of the buffer.

Could be solved by making `test_one_input` pass either a reference of a vector or a fuzzed data provider at compile time (with a compile flag)?

Feel free to ignore.

File: src/test/fuzz/FuzzedDataProvider.h [hunk 5]
```diff
@@ -0,0 +1,245 @@
+//===- FuzzedDataProvider.h - Utility header for fuzz targets ---*- C++ -* ===//
```
Comment (MarcoFalke) [inline src/test/fuzz/FuzzedDataProvider.h]: Could put this into a separate commit to explain in the commit body where it is taken from exactly (repo+commit)?

Comment (practicalswift) [reply src/test/fuzz/FuzzedDataProvider.h]: Good point! Fixed!

File: test/fuzz/test_runner.py [hunk 6]
```diff
@@ -12,6 +12,10 @@
 import subprocess
 import logging
 
+# Fuzzers known to lack a seed corpus in https://github.com/bitcoin-core/qa-assets/tree/master/fuzz_seed_corpus
+FUZZERS_MISSING_CORPORA = [
+    "eval_script",
```
Comment (MarcoFalke) [inline test/fuzz/test_runner.py]: can remove all of this?

Comment (practicalswift) [reply test/fuzz/test_runner.py]: Done! :)

File: src/test/fuzz/eval_script.cpp [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+#include <test/fuzz/FuzzedDataProvider.h>
+#include <test/fuzz/fuzz.h>
+
+#include <limits>
+
+void test_one_input(const std::vector<uint8_t>& buffer)
+{
+    if (buffer.size() <= sizeof(unsigned int)) {
+        return;
+    }
```
Comment (MarcoFalke) [inline src/test/fuzz/eval_script.cpp]: why is this needed? The fuzzer should be able to learn it trivially.

Also, you are limiting the script to have a size greater or equal to one

**Review by maflcko — COMMENTED** [review]
Concept ACK d58e0a5b39032e3dc06af0f0662b6a8ae092345d

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

ACK d58e0a5b39032e3dc06af0f0662b6a8ae092345d
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUhokAv9Gw2Cn+XfK88rrNb49RuOKqSslmk7/yvSEFZLTiA9U7guKJPBCeRFcMgV
ItO3JXlht3/6Ltot9/kt30EQDlwL/E2YTL7AFSv41FG9gvV8ORSVEZ1O4tS1pyx8
32Rgd7H34V5rg2qnKQR+zCCM2aBxtQ53wRQS0DjRKbS6mdUqky+rfZ2AwUUFwjeG
P9Vg/XMC6XkBMOvJqPS8f0Ef/YSJsjNTuDo2g6OzEbAd2UbF3eCR4aFD5Um8eW41
Dqm3YjB13UQwnqVSunazYtyiQauRJoxlkshmFec/taD4ZrIhDCCjuPiSRTE6zZgV
DKHukZO7OkYn0uVRimHirR0ezx06QuEp9wcbvv5L92Wecvfo2NlSHW9WSzvGwaV8
48agu4UTwEcTl8BgP+lTFjL3qncJ8W1z0Yjsbbgjg0/UMgSoqAAqeM9r2v5HqqGe
7NCdxO3RuoT2VKgU0IzslXolxtv5qyio9I041wkPW0pG/pZ4+qXn9sgQvMq9R1DR
X9AiIDKI
=ClLo
-----END PGP SIGNATURE-----
```

[Timestamp of file with hash `32f59337b799b6469837b88390d4a402e662f355ae288f53c5ff01c79d105ce1  -`](https://opentimestamps.org/info/?004f70656e54696d657374616d7073000050726f6f6600bf89e2e884e89294010832f59337b799b6469837b88390d4a402e662f355ae288f53c5ff01c79d105ce1f01067b53d85525920af51bb0f8402c2bf7808fff0105758956a5f293f932a8cf5a8bcb1a3d508f1045da09466f008f8322a2b05be3b810083dfe30d2ef90c8e232268747470733a2f2f6274632e63616c656e6461722e636174616c6c6178792e636f6dfff0107fb71b672e123145173e887f0eaf0d5a08f1045da09465f008680e0561d96d899c0083dfe30d2ef90c8e2c2b68747470733a2f2f626f622e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267f010ba39ce5076e17aa03d2f474b8ca46b0308f1045da09466f008d6d0f4007b880fb30083dfe30d2ef90c8e2e2d68747470733a2f2f616c6963652e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267)

</details>

File: src/test/fuzz/eval_script.cpp (hunk 7 above)
Comment (practicalswift) [reply src/test/fuzz/eval_script.cpp]: Unfortunately needed to not trigger the following UBSan warnings in Chromium's `FuzzedDataProvider.h` :)

Without:

```
$ src/test/fuzz/eval_script
INFO: Seed: 2056324753
INFO: Loaded 1 modules   (453398 inline 8-bit counters): 453398 [0x563c78576048, 0x563c785e4b5e),
INFO: Loaded 1 PC tables (453398 PCs): 453398 [0x563c785e4b60,0x563c78ccfcc0),
INFO: -max_len is not provided; libFuzzer will not generate inputs larger than 4096 bytes
test/fuzz/FuzzedDataProvider.h:212:17: runtime error: null pointer passed as argument 1, which is declared to never be null
/usr/include/string.h:43:28: note: nonnull attribute specified here
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior test/fuzz/FuzzedDataProvider.h:212:17 in
test/fuzz/FuzzedDataProvider.h:212:32: runtime error: null pointer passed as argument 2, which is declared to never be null
/usr/include/string.h:43:28: note: nonnull attribute specified here
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior test/fuzz/FuzzedDataProvider.h:212:32 in
```

Comment (MarcoFalke) [reply src/test/fuzz/eval_script.cpp]: Why doesn't that make the fuzzer fail and exit?

Comment (MarcoFalke) [reply src/test/fuzz/eval_script.cpp]: Anyway, the check should be `<`, not `<=`, otherwise you are limiting the script to have a size greater or equal to one

Comment (practicalswift) [reply src/test/fuzz/eval_script.cpp]: @MarcoFalke I think you forgot to set `UBSAN_OPTIONS=print_stacktrace=1:halt_on_error=1:exitcode=1` :)

I think this is the behaviour you expected:

```
$ UBSAN_OPTIONS=print_stacktrace=1:halt_on_error=1:exitcode=1 src/test/fuzz/eval_script -runs=1 ../btc_qa_assets/fuzz_seed_corpus/eval_script
INFO: Seed: 3606842015
INFO: Loaded 1 modules   (453402 inline 8-bit counters): 453402 [0x55b0f0056068, 0x55b0f00c4b82),
INFO: Loaded 1 PC tables (453402 PCs): 453402 [0x55b0f00c4b88,0x55b0f07afd28),
INFO:        0 files found in ../btc_qa_assets/fuzz_seed_corpus/eval_script
INFO: -max_len is not provided; libFuzzer will not generate inputs larger than 4096 bytes
test/fuzz/FuzzedDataProvider.h:212:17: runtime error: null pointer passed as argument 1, which is declared to never be null
/usr/include/string.h:43:28: note: nonnull attribute specified here
    #0 0x55b0eb940b1d in std::vector<unsigned char, std::allocator<unsigned char> > FuzzedDataProvider::ConsumeBytes<unsigned char>(unsigned long, unsigned long) /src/./test/fuzz/FuzzedDataProvider.h:212:5
    #1 0x55b0eb9404b7 in std::vector<unsigned char, std::allocator<unsigned char> > FuzzedDataProvider::ConsumeBytes<unsigned char>(unsigned long) /src/./test/fuzz/FuzzedDataProvider.h:42:12
    #2 0x55b0eb93e4c3 in std::vector<unsigned char, std::allocator<unsigned char> > FuzzedDataProvider::ConsumeRemainingBytes<unsigned char>() /src/./test/fuzz/FuzzedDataProvider.h:143:12
    #3 0x55b0eb93d70f in test_one_input(std::vector<unsigned char, std::allocator<unsigned char> > const&) /src/test/fuzz/eval_script.cpp:18:68
    #4 0x55b0eb9331fb in LLVMFuzzerTestOneInput /src/test/fuzz/fuzz.cpp:34:5
    #5 0x55b0eb6fbb87 in fuzzer::Fuzzer::ExecuteCallback(unsigned char const*, unsigned long) (/src/test/fuzz/eval_script+0x1da8b87)
    #6 0x55b0eb7057cb in fuzzer::Fuzzer::ReadAndExecuteSeedCorpora(std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, fuzzer::fuzzer_allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&) (/src/test/fuzz/eval_script+0x1db27cb)
    #7 0x55b0eb707932 in fuzzer::Fuzzer::Loop(std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, fuzzer::fuzzer_allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&) (/src/test/fuzz/eval_script+0x1db4932)
    #8 0x55b0eb6f6e1c in fuzzer::FuzzerDriver(int*, char***, int (*)(unsigned char const*, unsigned long)) (/src/test/fuzz/eval_script+0x1da3e1c)
    #9 0x55b0eb6bb2c2 in main (/src/test/fuzz/eval_script+0x1d682c2)
    #10 0x7f84c2817b96 in __libc_start_main /build/glibc-OTsEL5/glibc-2.27/csu/../csu/libc-start.c:310
    #11 0x55b0eb6e9d39 in _start (/src/test/fuzz/eval_script+0x1d96d39)

SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior test/fuzz/FuzzedDataProvider.h:212:17 in
MS: 0 ; base unit: 0000000000000000000000000000000000000000


artifact_prefix='./'; Test unit written to ./crash-da39a3ee5e6b4b0d3255bfef95601890afd80709
Base64:
$ echo $?
1
```

**Comment by MarcoFalke** [comment]
ACK 7e50abcc29dc5fde24a1b3e57c6316eabda35e2e

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

ACK 7e50abcc29dc5fde24a1b3e57c6316eabda35e2e
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUidXQwAq4GO999vR/OJASwj9LLmukMWUhvY04RwBHGUho2xHLi2V1YL1q/f5cYH
qVNb/V/llESB85HRz4kOYc+FntCFMD9IOFIAagLvYATqVk0n9iOdGHsbBovMtVUf
uBErkQLGfghU+b1hLxs0wmcNdY8KvmaI/o36GD2k03TZ+ZbkQ4F1UR4x6uaburv6
oo+pWtmgUZUFiUAZKnmz/DlNmURW05TlCdGgwcEPCum5v363lY8r9miVASEiezDi
EImqxL894G251U/j260Bc07RXiFeWS/9znNtYvwTZl8UNCCVNxrw0he3lG1ru7Z4
5aXwS6g8+rhzU38b+N+wM68X5NgrcQGpCAzCAafVvZHcKh4bqCikxId0fQ0xRMpA
Isy6KEQ8WVAoloelUubu6NUl0RhzJIAfEjD8cSXXdJ/oW8TLDHLxG3Kj0MjUKudw
7kNZxFgdrIl1e5kKlAhYVLxgCNiFWvqJo3uaruCMUpEw1mtxgvhcxqn/5+2keihM
6Tx7u40P
=UCP9
-----END PGP SIGNATURE-----
```

[Timestamp of file with hash `affe871723f783a800c6edcd8f373e476fd6c5791654096da973631289ee4718  -`](https://opentimestamps.org/info/?004f70656e54696d657374616d7073000050726f6f6600bf89e2e884e892940108affe871723f783a800c6edcd8f373e476fd6c5791654096da973631289ee4718f010108b2f2eef6d8b8ee0aa1cf4a810319c08fff01086c49f0819de519a2873e9f71681aa6d08f1045da477a9f008aecfbfaf8a2838ac0083dfe30d2ef90c8e292868747470733a2f2f66696e6e65792e63616c656e6461722e657465726e69747977616c6c2e636f6dfff010b87fde5004a9f386fb1404461c0af11408f1045da477a8f0083ef112ff93ec8ab60083dfe30d2ef90c8e2e2d68747470733a2f2f616c6963652e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267f010ef914908738fe13c4b0b9d86828edad208f1045da477a9f008328fcd4384a1f7080083dfe30d2ef90c8e2c2b68747470733a2f2f626f622e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267)

</details>
